### PR TITLE
Migrate integration_test plugin to Gradle Kotlin DSL

### PR DESCRIPTION
**Pull request by bartekpacia**
We already have a simple app in this repo that fully uses Gradle Kotlin DSL (see #141541). The next small step is to have at least one plugin that also uses Gradle Kotlin DSL. Let's use `integration_test` for that, since it's versioned with Flutter SDK.

### CODE REVIEW

**Comment by bartekpacia** [comment]
Currently it only supports AGP v8+ and Gradle v8+. I'll try to get down to v7+ for both. Related #142000

File: packages/integration_test/android/build.gradle [hunk 1]
```diff
@@ -1,64 +0,0 @@
...
-//     options.compilerArgs << "-Xlint:all" << "-Werror"
-// }
-
-apply plugin: 'com.android.library'
-
-android {
-    // Conditional for compatibility with AGP <4.2.
-    if (project.android.hasProperty("namespace")) {
-        namespace 'dev.flutter.integration_test'
-    }
```
Comment (bartekpacia) [inline packages/integration_test/android/build.gradle:36]: PR #142000 will start enforcing Gradle 7, so this checks won't make sense anymore

Comment (gmackall) [reply packages/integration_test/android/build.gradle:36]: Heads up that the linked PR actually won't start enforcing anything until a full release later. I'm starting with only "warning" versions, so when it lands we will start warning that anything <7 _will soon_ be unsupported

Comment (bartekpacia) [reply packages/integration_test/android/build.gradle:36]: ah okay, thanks! So I'll make sure we support older Gradle/AGP as well

**Comment by reidbaker** [comment]
The last migration just was reverted. Please dont land this pr until we know why the previous one broke the tree. 
https://github.com/flutter/flutter/pull/142018#event-11563334311

**Comment by gmackall** [comment]
> The last migration was just reverted. Please dont land this pr until we know why the previous one broke the tree. [#142018 (comment)](https://github.com/flutter/flutter/pull/142018#event-11563334311)

I see the migration which was reverted was relanded in https://github.com/flutter/flutter/pull/142752, but I can't see why it was reverted or what the fix was - can you add any more context @bartekpacia ? Just in case it is relevant here as well

**Review by gmackall — COMMENTED** [review]
LGTM but just holding off on approving till I know why the previous migration was reverted

**Comment by bartekpacia** [comment]
@gmackall The reason for reverting the original "Add support Gradle Kotlin DSL" PR (#140744) is because a a different PR (#142266) [had to be reverted](142266), and it was easiest to do it by also reverting my original "Add support Gradle Kotlin DSL" PR (#140744).

Now the situation is sorted out, and the new "Add support Gradle Kotlin DSL" PR (#142752)has been merged for a few days and tree is still green.

**Comment by gmackall** [comment]
Got it, thanks!

**Review by gmackall — APPROVED** [review]

**Comment by bartekpacia** [comment]
I've got a bad feeling that something will break post-submit, haha.

**Comment by camsim99** [comment]
> I've got a bad feeling that something will break post-submit, haha.

😭 @gmackall pinged me because it looks like this is breaking the Flutter --> packages roller, e.g. https://github.com/flutter/packages/pull/6105:

```
e: /b/s/w/ir/cache/gradle/caches/jars-8/42139b5be0c79e133a7b5c5c5d5bf731/gradle-7.3.1.jar!/com/android/build/gradle/LibraryExtension.class: Class 'com/android/build/gradle/LibraryExtension' was compiled with an incompatible version of Kotlin. The binary version of its bytecode is unknown, expected version is 1.0.3
e: /b/s/w/ir/cache/gradle/caches/jars-8/043c0d54e2c1bb39208ec4477607cdf5/gradle-api-7.3.1.jar!/com/android/build/api/dsl/LibraryVariantDimension.class: Class 'com/android/build/api/dsl/LibraryVariantDimension' was compiled with an incompatible version of Kotlin. The binary version of its bytecode is unknown, expected version is 1.0.3

FAILURE: Build failed with an exception.

* Where:
Build file '/b/s/w/ir/x/w/flutter/packages/integration_test/android/build.gradle.kts' line: 36

* What went wrong:
com.android.build.gradle.LibraryExtension.setNamespace(Ljava/lang/String;)V

```

Gonna try to auto-revert, and if not open a manual one

**Comment by camsim99** [comment]
Reason for revert: breaking Flutter -> packages roller

**Pull request merged**